### PR TITLE
Add unit tests for GradientTexture

### PR DESCRIPTION
**Pull request by ArthyChaux**
Hello,
This PR adds unit tests for GradientTexture mentioned in #43440. The tests may be incomplete.

It should also work on the 3.x branch

### CODE REVIEW

File: tests/scene/test_gradient_texture.h [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+			"Error in fill to vector method.");
+
+	gradient_texture->set_repeat(GradientTexture2D::Repeat::REPEAT);
+	CHECK_MESSAGE(
+			gradient_texture->get_repeat() == GradientTexture2D::Repeat::REPEAT,
+			"Error in repeat setter or getter.");
+}
+
+} //namespace TestGradientTexture
+#endif // TEST_GRADIENT_TEXTURE_H
```
Comment (AThousandShips) [inline tests/scene/test_gradient_texture.h:109]:
```suggestion

#endif // TEST_GRADIENT_TEXTURE_H

```

**Comment by AThousandShips** [comment]
Please squash your commits into one, see [here](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase), and give your single commit a clear name, the name of the PR is good

File: tests/scene/test_gradient_texture.h [hunk 2]
```diff
@@ -0,0 +1,109 @@
+/**************************************************************************/
+/*  test_gradient.h                                                       */
```
Comment (AThousandShips) [inline tests/scene/test_gradient_texture.h:2]:
```suggestion
/*  test_gradient_texture.h                                               */
```

File: tests/scene/test_gradient_texture.h [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+			"Error in Use HDR setter or getter.");
+
+	gradient_texture->set_fill(GradientTexture2D::Fill::FILL_SQUARE);
+	CHECK_MESSAGE(
+			gradient_texture->get_fill() == GradientTexture2D::Fill::FILL_SQUARE,
+			"Error in fill method.");
+
+	gradient_texture->set_fill_from(vector);
+	CHECK_MESSAGE(
+			gradient_texture->get_fill_from().operator==(vector),
```
Comment (AThousandShips) [inline tests/scene/test_gradient_texture.h:94]:
```suggestion
			gradient_texture->get_fill_from() == vector,
```

File: tests/scene/test_gradient_texture.h [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+			"Error in fill from vector method.");
+
+	gradient_texture->set_fill_to(vector);
+	CHECK_MESSAGE(
+			(gradient_texture->get_fill_to().operator==(vector)),
```
Comment (AThousandShips) [inline tests/scene/test_gradient_texture.h:99]:
```suggestion
			gradient_texture->get_fill_to() == vector,
```

**Comment by akien-mga** [comment]
Superseded by #90501. Thanks for the contribution!

**Pull request closed without merging**